QuizApp/QuizApp/UI/Screens/QuizDetails/QuizDetailsViewController.swift

```diff
@@ -1,30 +1,45 @@
+import Combine
 import UIKit
 
 class QuizDetailsViewController: UIViewController {
 
     private var appRouter: AppRouterProtocol!
 
-    private var quiz: Quiz!
+    private var quizDetailsViewModel: QuizDetailsViewModel!
 
     private var gradientView: GradientView!
+
     private var mainView: UIView!
+
     private var detailsView: DetailsView!
 
-    init(appRouter: AppRouterProtocol, quiz: Quiz) {
+    private var cancellables = Set<AnyCancellable>()
+
+    init(appRouter: AppRouterProtocol, quizDetailsViewModel: QuizDetailsViewModel) {
```

Comment:
Why do you need router in VC? VM should call the router to navigate. Also, you can shorten the field/argument name to viewModel, this VC has just one VM.